addons/base_import/models/base_import.py

```diff
@@ -1627,11 +1638,24 @@ def _handle_multi_mapping(self, import_fields, input_file_data):
 
                 # merge data if necessary
                 if field_type == 'char':
-                    new_record.append(' '.join(record[idx] for idx in indexes if record[idx]))
+                    new_record.append(
+                        ' '.join(
+                            self._stringify_date_like_objects(record[idx], options=self.env.context.get('options', {}))
```

Comment:
Can you put `self.env.context.get('options', {})` at the begin of the method to avoid get + create one dict for each record.